pkg/oc/cli/cmd/rollout/cancel.go

```diff
@@ -136,7 +136,14 @@ func (o CancelOptions) Run() error {
 				return runtime.Encode(o.Encoder, rc)
 			})
 
-			if len(patches) == 0 {
+			allPatchesEmpty := true
+			for _, patch := range patches {
+				if len(patch.Patch) > 0 {
```

Comment:
can there be patch '{}'? otherwise lgtm except the debugs

Reply:
> can there be patch '{}'? otherwise lgtm except the debugs

I don't think that produces the same error. I'm actually wondering if the server changed to reject empty patches...

Reply:
other places use constructs like this
```go
if len(patch.Patch) > 0 && string(patch.Patch) != "{}" 
```
https://github.com/openshift/origin/blob/master/pkg/oc/cli/cmd/set/triggers.go#L332-L335

Reply:
`CreateTwoWayMergePatch` looks sane in 1.8 and 1.9, both giving `{}´ for no changes.